fs-cache/src/memory_limited_storage.rs

```diff
@@ -0,0 +1,273 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use data_error::{ArklibError, Result};
+use linked_hash_map::LinkedHashMap;
+
+pub struct MemoryLimitedStorage<K, V> {
+    /// Label for logging
+    label: String,
+    /// Path to the underlying folder where data is persisted
+    path: PathBuf,
+    /// In-memory LRU cache combining map and queue functionality
+    memory_cache: LinkedHashMap<K, V>,
+    // Bytes present in memory
+    current_memory_bytes: usize,
+    /// Maximum bytes to keep in memory
+    max_memory_bytes: usize,
+}
+
+impl<K, V> MemoryLimitedStorage<K, V>
+where
+    K: Ord
+        + Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::fmt::Display
+        + std::hash::Hash
+        + std::str::FromStr,
+    V: Clone + serde::Serialize + serde::de::DeserializeOwned,
+{
+    pub fn new(
+        label: String,
+        path: &Path,
+        max_memory_bytes: usize,
+    ) -> Result<Self> {
+        let mut storage = Self {
+            label,
+            path: PathBuf::from(path),
+            memory_cache: LinkedHashMap::new(),
+            current_memory_bytes: 0,
+            max_memory_bytes,
+        };
+
+        storage.load_fs()?;
+
+        Ok(storage)
+    }
+
+    pub fn label(&self) -> String {
+        self.label.clone()
+    }
+
+    pub fn get(&mut self, key: &K) -> Option<V> {
+        // Check memory cache first - will update LRU order automatically
+        if let Some(value) = self.memory_cache.get_refresh(key) {
```

Comment:
That's pretty cool, I didn't know about `get_refresh` function.

But I guess it works with _O(N)_ complexity, right? Consider using dedicated Rust crate for LRU to get _O(1)_ complexity. This isn't pressing issue, we can create an issue for it.